src/core/LadspaControl.cpp

```diff
@@ -342,16 +342,18 @@ void LadspaControl::unlinkControls( LadspaControl * _control )
 	switch( m_port->data_type )
 	{
 		case BufferDataType::Toggled:
-			BoolModel::unlinkModels( &m_toggledModel, _control->toggledModel() );
+			m_toggledModel.unlinkAllModels();
+			_control->toggledModel()->unlinkAllModels();
```

Comment:
Don't both models exist in the same linked list, so shouldn't you only need to call one of these? (Same goes for the other lines below this)

Reply:
@messmerd with the latest push this is no longer true.

The code we're looking at is the one for effects which have a LED checkbox to link L-R channels. Imagine this:
```
Attack    (L) (R)     [x] link
Delay     (L) (R)     [x] link
```

By default L-R channels are linked. If you then ctrl-drag Attack-L to Delay-L all four knobs in this example are linked. If you uncheck the link on Attack then its L-R models will be unlinked. **But it should not call `unlinkAllModels()` on *both* L and R.** That way the user-made link between Attack and Delay will persist on at least one of the knobs. So when the user checks the link checkbox again all four knobs will once again be linked.